test/com/xilinx/rapidwright/design/merge/TestMergeDesigns.java

```diff
@@ -0,0 +1,189 @@
+            }
+            
+        }
+        if(reverse) {
+            Design tmp = designs.getFirst();
+            designs.setFirst(designs.getSecond());
+            designs.setSecond(tmp);
+        }
+        
+        return designs;
+    }
+
+    private static int countMergedNets(EDIFCell ... topCells) {
+        HashSet<String> nets = new HashSet<>();
+        for(EDIFCell top : topCells) {
+            for(EDIFNet net : top.getNets()) {
+                nets.add(net.getName());
+            }
+        }
+        return nets.size();
+    }
+    
+    private static int countMergedInsts(EDIFCell ... topCells) {
+        HashSet<String> insts = new HashSet<>();
+        for(EDIFCell top : topCells) {
+            for(EDIFCellInst inst : top.getCellInsts()) {
+                insts.add(inst.getName());
+            }
+        }
+        return insts.size();
+    }
+    
+    private static int countMergedPorts(EDIFCell ... topCells) {
+        HashSet<String> ports = new HashSet<>();
+        for(EDIFCell top : topCells) {
+            for(EDIFPort port : top.getPorts()) {
+                ports.add(port.getName());
+            }
+        }
+        return ports.size();
+    }
+    
+    private static int countMergedSiteInsts(Design ... designs) {
+        HashSet<String> sites = new HashSet<>();
+        for(Design design : designs) {
+            for(SiteInst siteInst : design.getSiteInsts()) {
+                if(siteInst.isPlaced()) {
+                    sites.add(siteInst.getSiteName());
+                }
+            }
+        }
+        return sites.size();
+    }
+    
+    @Test
+    @CheckOpenFiles
+    public void testMergePicoblaze() throws IOException {
+        Design design0 = Design.readCheckpoint(RapidWrightDCP.getPath("picoblaze_ooc_X10Y235.dcp"));
+        Design design1 = Design.readCheckpoint(RapidWrightDCP.getPath("picoblaze4_ooc_X6Y60_X6Y65_X10Y60_X10Y65.dcp"));
+        EDIFCell top0 = design0.getTopEDIFCell();
+        EDIFCell top1 = design1.getTopEDIFCell();
+        
+        Design merged = MergeDesigns.mergeDesigns(design0, design1);
+        
+        EDIFCell top = merged.getTopEDIFCell();
+
+        // Check the merged 'reset' net
+        String reset = "reset";
+        Assertions.assertEquals(top.getNet(reset).getPortInsts().size(), 
+                                top0.getNet(reset).getPortInsts().size() + 
+                                top1.getNet(reset).getPortInsts().size() - 1);
+        
+        Assertions.assertEquals(top.getCellInsts().size(), countMergedInsts(top0, top1));
+        Assertions.assertEquals(top.getNets().size(), countMergedNets(top0, top1));
+        Assertions.assertEquals(top.getPorts().size(), countMergedPorts(top0, top1));
+        Assertions.assertEquals(merged.getSiteInsts().size(), countMergedSiteInsts(design0, design1));
+    }
+    
+    @ParameterizedTest
+    @ValueSource(strings = {"False", "True"}) // Doesn't support boolean :(
```

Comment:
Did you try this? https://stackoverflow.com/a/59958483

Reply:
Huh, I guess some dude on the Internet was wrong.